client/txn.go

```diff
@@ -0,0 +1,135 @@
+package client
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/google/uuid"
+	xlineapi "github.com/xline-kv/go-xline/api/xline"
+)
+
+type Txn interface {
+	// If takes a list of comparison. If all comparisons passed in succeed,
+	// the operations passed into Then() will be executed. Or the operations
+	// passed into Else() will be executed.
+	If(cs ...Cmp) Txn
+
+	// Then takes a list of operations. The Ops list will be executed, if the
+	// comparisons passed in If() succeed.
+	Then(ops ...Op) Txn
+
+	// Else takes a list of operations. The Ops list will be executed, if the
+	// comparisons passed in If() fail.
+	Else(ops ...Op) Txn
+
+	// Commit tries to commit the transaction.
+	Commit() (*TxnResponse, error)
+}
+
+type txn struct {
+	// Name of the kv client, which will be used in CURP propose id generation
+	name string
+	// The client running the CURP protocol, communicate with all servers.
+	curpClient curpClient
+	// The auth token
+	token string
+
+	mu    sync.Mutex
+	cif   bool
+	cthen bool
+	celse bool
+
+	cmps []*xlineapi.Compare
+	sus  []*xlineapi.RequestOp
+	fas  []*xlineapi.RequestOp
+}
+
+func (txn *txn) If(cs ...Cmp) Txn {
```

Comment:
Could we allow `If(cmp Cmp) Txn` multiple times instead of call `If(cs ...Cmp) Txn` once?

Reply:
We can't allow call `If(cmp Cmp) Txn` multiple times, so we write this:
```go
if txn.cif {
  panic("cannot call If twice!")
}
```